lighthouse-core/gather/gatherers/dobetterweb/link-in-head.js

```diff
@@ -0,0 +1,96 @@
+/**
+ * @license
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const Gatherer = require('../gatherer');
+
+/* global document,window */
+
+/* istanbul ignore next */
+function getBlockFirstPaintLink() {
+// filtered match stylesheet/import
+// ref)
+// https://www.igvita.com/2012/06/14/debunking-responsive-css-performance-myths/
+// https://www.w3.org/TR/html-imports/#dfn-import-async-attribute
+  const linkList = [... document.querySelectorAll('head link')].filter(link => {
+    const asyncAttr = link.getAttribute('async');
+    return (link.rel === 'stylesheet' && window.matchMedia(link.media).matches) ||
+           (link.rel === 'import' && (asyncAttr === '' || asyncAttr));
```

Comment:
using `link.hasAttribute('async')` might be nicer. Then you can just check against a boolean.